USF-4113: document dropins MCP server and storefront skills for AEM Boilerplate Commerce (#576)
AdobeDocs/commerce-extensibility: src/pages/developer-agent/skills-and-prompts.md

| `/technical-writer` | Creates comprehensive documentation for App Builder applications. Use when writing `README.md`, user guides, API documentation, changelogs, or ensuring documentation completeness. |
| `/tester` | Creates comprehensive tests for App Builder applications. Use when writing unit tests, integration tests, validating security, or ensuring code quality and coverage. |
| `/tutor` (experimental) | Teaches Adobe Commerce application development concepts with clear explanations and examples. Use when learning App Builder, understanding events, or needing guidance on development patterns. |

## Storefront skills

> When you select the **AEM Boilerplate Commerce** starter kit during [setup](coding-tools.md#dropins-mcp-server-for-storefront-development), a separate set of skills is installed for storefront development. These skills route drop-in data queries (slots, events, containers, API functions) to the `dropins` MCP server as the primary source, falling back to TypeScript definitions in `node_modules/@dropins/` when the server is unavailable.

Reviewer: ```suggestion
When you select the **AEM Boilerplate Commerce** starter kit during [setup](coding-tools.md#dropins-mcp-server-for-storefront-development), a separate set of skills is installed for storefront development. These skills route drop-in data queries (slots, events, containers, and API functions) to the `dropins` MCP server as the primary reference, and fallback to the TypeScript definitions in `node_modules/@dropins/` when the server is unavailable.
```

| Command | Purpose |
| --- | --- |
| `/project-manager` | Gathers storefront requirements, analyzes the current project state with `dropins:analyze_project`, and manages the phased development protocol. |

> | `/researcher` | Looks up authoritative drop-in data — slots, events, containers, API functions, models, design tokens, and i18n keys — before any implementation work. |

Reviewer: ```suggestion
| `/researcher` | Looks up authoritative drop-in data for slots, events, containers, API functions, models, design tokens, and i18n keys before any implementation work. |
```

> | `/dropin-developer` | Scaffolds blocks, slots, and checkout extensions wired to real drop-in containers, and suggests slot implementations and event handlers. |

Reviewer: ```suggestion
| `/dropin-developer` | Scaffolds blocks, slots, and checkout extensions wired to real drop-in containers, and then suggests slot implementations and event handlers. |
```

| `/block-developer` | Implements and modifies AEM blocks that consume drop-in containers, following the boilerplate's file structure and conventions. |
| `/content-modeler` | Models content structures and authoring patterns for AEM Universal Editor blocks. |
| `/tester` | Runs pre-test health checks (`dropins:check_config`, `dropins:check_block_health`, `dropins:get_upgrade_diff`) and validates storefront behavior in the browser. |

### `dropins` MCP tools

The storefront skills call the following `dropins` MCP tools directly. You can also invoke them yourself in a prompt.

| Need to know | MCP tool |
| --- | --- |
| Slot names and context shapes for a container | `dropins:list_slots` |
| Event names, payloads, producers, consumers | `dropins:list_events` |
| Container names and their props | `dropins:list_containers` |
| Public API function signatures | `dropins:list_api_functions` |
| Shared model/type definitions (CartModel, Price, etc.) | `dropins:list_models` |
| CSS design tokens by category | `dropins:list_design_tokens` |
| i18n dictionary keys and defaults | `dropins:list_i18n_keys` |

> | Available GraphQL operations | `dropins:list_graphql_queries` |

Reviewer: ```suggestion
| Lists slot names and context shapes for a container | `dropins:list_slots` |
| Lists event names, payloads, producers, and consumers | `dropins:list_events` |
| Lists container names and their props | `dropins:list_containers` |
| Lists public API function signatures | `dropins:list_api_functions` |
| Lists shared model/type definitions, such as `CartModel` or `Price`) | `dropins:list_models` |
| Lists CSS design tokens by category | `dropins:list_design_tokens` |
| Lists i18n dictionary keys and defaults | `dropins:list_i18n_keys` |
| Lists available GraphQL operations | `dropins:list_graphql_queries` |
```

| Event emitters, consumers, and sample code | `dropins:explain_event_flow` |
| Keyword search across storefront docs | `dropins:search_docs` |
| Block health — outdated container/slot/API references | `dropins:check_block_health` |
| Storefront config validation (endpoints, environment) | `dropins:check_config` |
| Drop-in version gaps vs. the registry | `dropins:get_upgrade_diff` |
| Installed drop-ins, block inventory, version state | `dropins:analyze_project` |

> | Scaffold a block, slot, or checkout extension | `dropins:scaffold_block` / `dropins:scaffold_slot` / `dropins:scaffold_extension` |

Reviewer: ```suggestion
| Explains event emitters, consumers, and sample code | `dropins:explain_event_flow` |
| Performs a keyword search across storefront docs | `dropins:search_docs` |
| Checks the block health and indicates any outdated container, slot, or API references | `dropins:check_block_health` |
| Validates the storefront configuration, including endpoints and environment settings | `dropins:check_config` |
| Performs a diff comparing drop-in version gaps and the registry | `dropins:get_upgrade_diff` |
| Analyzes installed drop-ins, block inventory, and version state | `dropins:analyze_project` |
| Scaffolds a block, slot, or checkout extension | `dropins:scaffold_block` / `dropins:scaffold_slot` / `dropins:scaffold_extension` |
```
